src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -197,13 +197,27 @@ public DynamoView(DynamoViewModel dynamoViewModel)
             // Add an event handler to check if the collection is modified.   
             TabItems.CollectionChanged += this.OnCollectionChanged;
 
+            HomeWorkspaceModel.CloseExtension += this.CloseExtensionTab;
+
             this.HideOrShowRightSideBar();
 
             this.dynamoViewModel.RequestPaste += OnRequestPaste;
             this.dynamoViewModel.RequestReturnFocusToView += OnRequestReturnFocusToView;
             FocusableGrid.InputBindings.Clear();
         }
 
+        /// <summary>
+        /// This method is called when the home workspace is closed. This will inturn close the workspace references extension.
+        /// </summary>
+        /// <param name="extensionTabName">Extension name</param>
+        /// <returns></returns>
+        internal void CloseExtensionTab(String extensionTabName)
+        {
+            TabItem tabitem = TabItems.OfType<TabItem>().SingleOrDefault(n => n.Header.ToString() == extensionTabName);
```

Comment:
I really think this is a bad name for this control:
`TabItems` - because the DynamoView also has another set of TabItems for the workspaces...

Reply:
Should I rename this to ExtensionTabItems so that it would be clear to differentiate?

Reply:
This is renamed in https://github.com/DynamoDS/Dynamo/pull/10279

Reply:
I have pulled in your changes and this PR will be merged on top of #10279. So it should be fine.